framework/web/JsonParser.php

```diff
@@ -39,16 +39,13 @@ class JsonParser implements RequestParserInterface
 
 
     /**
-     * Parses a HTTP request body.
-     * @param string $rawBody the raw HTTP request body.
-     * @param string $contentType the content type specified for the request body.
-     * @return array parameters parsed from the request body
+     * {@inheritdoc}
      * @throws BadRequestHttpException if the body contains invalid json and [[throwException]] is `true`.
      */
-    public function parse($rawBody, $contentType)
+    public function parse($request)
     {
         try {
-            $parameters = Json::decode($rawBody, $this->asArray);
+            $parameters = Json::decode($request->getBody()->__toString(), $this->asArray);
```

Comment:
> $request->getBody()->__toString()

Request has method getRawBody()

Reply:
This method usage should be avoided.
Since PSR-7 accepted its interface should be used as wide as possible.